nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -449,6 +449,12 @@ def on_validation_batch_end(self, outputs, batch: Any, batch_idx: int, dataloade
     def setup_optimization(
         self, optim_config: Optional[Union[DictConfig, Dict]] = None, optim_kwargs: Optional[Dict[str, Any]] = None,
     ):
+        # Ensure `max_steps` is set correctly
+        optim_config = self._get_optim_config_copy(optim_config)
```

Comment:
This creates a copy of the config, and another one is also created in `ModelPT` ==> we end up copying it twice. It's a bit unsatisfying but since it's a very small config the performance hit is negligible, and trying to avoid the extra copy would probably make the code more compelx.